### PR TITLE
Support TFDS format for segmentation

### LINKED ISSUE #1002: Support TFDS format for Segmentation
Supporting TFDS format is good for training on cloud storages (e.g. GCS).
Current Blueoil supports TFDS format for classification and object detection, but not segmentation. I think it is better to support it!

### PR DESCRIPTION
**Pull request by hadusam**
## What this patch does to fix the issue.
Close #1002 

## Refs
I referred to the official tutorial of segmentation to decide the TFDS feature format. (`image`, `label`, `segmentation_mask`)
https://www.tensorflow.org/tutorials/images/segmentation#download_the_oxford-iiit_pets_dataset
```
tfds.core.DatasetInfo(
    name='oxford_iiit_pet',
    version=3.0.0,
    description='The Oxford-IIIT pet dataset is a 37 category pet image dataset with roughly 200
images for each class. The images have large variations in scale, pose and
lighting. All images have an associated ground truth annotation of breed.
',
    urls=['http://www.robots.ox.ac.uk/~vgg/data/pets/'],
    features=FeaturesDict({
        'file_name': Text(shape=(), dtype=tf.string),
        'image': Image(shape=(None, None, 3), dtype=tf.uint8),
        'label': ClassLabel(shape=(), dtype=tf.int64, num_classes=37),
        'segmentation_mask': Image(shape=(None, None, 1), dtype=tf.uint8),
    }),
    total_num_examples=7349,
    splits={
        'test': 3669,
        'train': 3680,
    },
    supervised_keys=('image', 'label'),
    citation="""@InProceedings{parkhi12a,
      author       = "Parkhi, O. M. and Vedaldi, A. and Zisserman, A. and Jawahar, C.~V.",
      title        = "Cats and Dogs",
      booktitle    = "IEEE Conference on Computer Vision and Pattern Recognition",
      year         = "2012",
    }""",
    redistribution_info=,
)
```

### CODE REVIEW

File: blueoil/utils/tfds_builders/segmentation.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        self.dataset_kwargs = dataset_kwargs
+        super().__init__(**kwargs)
+
+    def _info(self):
+        return tfds.core.DatasetInfo(
+            builder=self,
+            description="Custom TFDS dataset for segmentation",
+            features=tfds.features.FeaturesDict({
+                "image": tfds.features.Image(),
+                "label": tfds.features.ClassLabel(),
```
Comment (tfujiwar) [inline blueoil/utils/tfds_builders/segmentation.py:42]: So the feature `label` is used for having a list of classes but not needed for every example. Is my understanding correct?

Comment (hadusam) [reply blueoil/utils/tfds_builders/segmentation.py:42]: @tfujiwar 
Yes! Your understanding is correct! 👍

Comment (tfujiwar) [reply blueoil/utils/tfds_builders/segmentation.py:42]: Thank you!

**Comment by hadusam** [comment]
I checked and compared trainings with TFDS and without TFDS(local image files on directories) with my sample config. (CamVid with IMAGE_SIZE = [160, 240])
This looks the same between them 👍
 
image

**Pull request merged**